crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -1219,7 +1219,13 @@ impl<TX: DbTx> BlockNumReader for DatabaseProvider<TX> {
     }
 
     fn last_block_number(&self) -> ProviderResult<BlockNumber> {
-        Ok(self.tx.cursor_read::<tables::CanonicalHeaders>()?.last()?.unwrap_or_default().0)
+        Ok(match self.tx.cursor_read::<tables::CanonicalHeaders>()?.last()? {
+            Some((num, _)) => num,
+            None => self
+                .static_file_provider
+                .get_highest_static_file_block(StaticFileSegment::Headers)
+                .unwrap_or_default(),
+        })
```

Comment:
unless something went wrong/theres a bug, this should always unwrap successfully. But just in case, should we panic/error if it cant?